tests/TestFailureTest.cpp

```diff
@@ -333,6 +333,38 @@ TEST(TestFailure, BinaryEqualFullWidth)
     			"\t                                               ^", f);
 }
 
+TEST(TestFailure, IntsEqualOneByte)
+{
+    const unsigned char expectedData[] = { 0x01 };
+    const unsigned char actualData[] = { 0x02 };
+    AnyIntsEqualFailure f(test, failFileName, failLineNumber, expectedData, actualData, sizeof(expectedData), "");
+    FAILURE_EQUAL("expected <1 0x1>\n\tbut was  <2 0x2>", f);
+}
+
+TEST(TestFailure, IntsEqualTwoBytes)
+{
+    const unsigned char expectedData[] = {0x00, 0x01};
+    const unsigned char actualData[] = {0x00, 0x02};
+    AnyIntsEqualFailure f(test, failFileName, failLineNumber, expectedData, actualData, sizeof(expectedData), "");
+    FAILURE_EQUAL("expected <256 0x100>\n\tbut was  <512 0x200>", f);
+}
+
+TEST(TestFailure, IntsEqualFourBytes)
+{
+    const unsigned char expectedData[] = {0x00, 0x01, 0x00, 0x00};
+    const unsigned char actualData[] = {0x00, 0x02, 0x00, 0x00};
+    AnyIntsEqualFailure f(test, failFileName, failLineNumber, expectedData, actualData, sizeof(expectedData), "");
+
+    if (sizeof(long) >= 4)
```

Comment:
1. In my mind, the first part of the "if" just means making a bad result okay. If I understand correctly what you are doing, then the outcome of your comparison should be the same, irrespective of size of long? After all, you are passing in the integer data as an array of bytes?

2. There  is no test for 64 bit or 128 bit.

3. An int can be negative. I don't think your methods can handle negative values (yet?).

4. What about little-endian vs. big-endian?